cloud_images/lib/el7/configure_dcos_system.sh

```diff
@@ -18,23 +18,36 @@ sed -i '/^\s*UseDNS /d' /etc/ssh/sshd_config
 echo -e "\nUseDNS no" >> /etc/ssh/sshd_config
 
 echo ">>> Set up filesystem mounts"
-cat << 'EOF' > /etc/systemd/system/dcos_vol_setup.service
-[Unit]
-Description=Initial setup of volume mounts
-DefaultDependencies=no
-Before=local-fs-pre.target
-
-[Service]
-Type=oneshot
-ExecStart=/usr/local/sbin/dcos_vol_setup.sh /dev/xvde /var/lib/mesos
-ExecStart=/usr/local/sbin/dcos_vol_setup.sh /dev/xvdf /var/lib/docker
-ExecStart=/usr/local/sbin/dcos_vol_setup.sh /dev/xvdg /dcos/volume0
-ExecStart=/usr/local/sbin/dcos_vol_setup.sh /dev/xvdh /var/log
-
-[Install]
-RequiredBy=local-fs-pre.target
+mount_pairs=( "/dev/xvde:/var/lib/mesos"
+              "/dev/nvme1n1:/var/lib/mesos"
+              "/dev/xvdf:/var/lib/docker"
+              "/dev/nvme2n1:/var/lib/docker"
+              "/dev/xvdg:/dcos/volume0"
+              "/dev/nvme3n1:/dcos/volume0"
+              "/dev/xvdh:/var/log"
+              "/dev/nvme4n1:/var/log"
+            )
+for mount_pair in ${mount_pairs[@]}; do
+  device=$(echo ${mount_pair} | cut -d':' -f1)
+  mountpoint=$(echo ${mount_pair} | cut -d':' -f2)
+  device_filenamesafe=$(echo ${device} | sed 's/\//-/g')
+
+  cat << EOF > /etc/systemd/system/dcos_vol_setup${device_filenamesafe}.service
+  [Unit]
+  Description=Initial setup of volume mounts
+  DefaultDependencies=no
+  Before=local-fs-pre.target
+
+  [Service]
+  Type=oneshot
+  TimeoutSec=20
```

Comment:
Good point! Thanks for bringing it up, I struggled to see any case in which a migration to a new EBS volumes happens after the machine ran for an extended period of time. To my understanding it will run at the very first boot, migrating the very few existing log files from the base AMI and the running boot process. Which cases did I miss?

Reply:
I'm just wondering if there is a reason to introduce the reduced 20 second timeout as opposed to the 90 second default (unless changed by the user via DefaultTimeoutStartSec)? Users are using these scripts in their own environments in ways we don't control and might be different from how we're using them. Before making any changes that could break those environments I'd want to be sure there's a technical necessity for it. I.e. what problem are we solving by making the volume setup abort 70 seconds sooner?